src/query.ts

```diff
@@ -238,4 +238,27 @@ export class Query extends Builder<Query> {
     this.clauses.addClause(clause);
     return this;
   }
+
+  /**
+   * Adds a new parameter to the query
+   *
+   * @param {any} Value
+   * @param {string} Name of the parameter
+   * @returns {this}
+   */
+  addParam(value: any, name: string): this {
```

Comment:
Would we be able to make the name parameter optional here, just so it is consistent with the method in `ParameterContainer` and `ParameterBag`. 

Also I see how returning `this` is consistent with the rest of the functions in the query interface, however, one of the things about parameters is that their name may be different from what is given if a parameter with that name already exists. For example `addParam('...', 'name')` may actually add a parameter with the name `name2` if a parameter with the name `name` already exists. In those cases it would be useful to have the parameter object returned from this function to know exactly what name the parameter was given.

Reply:
@jamesfer Sure. Also, I would maybe add a third param (bool) to force keeping the param name (override the current value)? 

The only inconvenient with not returning `this` is that you can no longer chain it. 
However, my use case does not require that (otherwise I would just use `raw()`).

Reply:
Yeah the not returning `this` is inconvenient, but I hope that people don't need to use this function too often.